Sources/Model/Paging/HyperwalletReceiptQueryParam.swift

```diff
@@ -0,0 +1,86 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+/// Representation of the user receipts pagination fields.
+public class HyperwalletReceiptQueryParam: HyperwalletQueryParam {
+    /// Returns transfer methods created after this datetime.
+    public var createdAfter: Date?
+    /// Returns transfer methods created before this datetime.
+    public var createdBefore: Date?
+    /// A value that identifies the transfer method currency.
+    public var currency: String?
+    /// The transfer methods attribute to sort the result set by.
+    public var sortBy: QuerySortable?
+
+    private enum QueryParam: String {
+        case createdAfter
+        case createdBefore
+        case currency
+        case sortBy
+    }
+
+    /// Representation of the field's sortable
+    ///
+    /// - ascendantAmount: Sort the result by ascendant the field amount
+    /// - ascendantCreatedOn: Sort the result by ascendant the field create on
+    /// - ascendantCurrency: Sort the result by ascendant the field currency
+    /// - ascendantType: Sort the result by ascendant the field type
+    /// - descendantAmount: Sort the result by descendant the create amount
+    /// - descendantCreatedOn: Sort the result by descendant the field created on
+    /// - descendantCurrency: Sort the result by descendant the field currency
+    /// - descendantType: Sort the result by descendant the field type
+    public enum QuerySortable: String {
+        case ascendantAmount = "+amount"
+        case ascendantCreatedOn = "+createdOn"
+        case ascendantCurrency = "+currency"
+        case ascendantType = "+type"
+
+        case descendantAmount = "-amount"
+        case descendantCreatedOn = "-createdOn"
+        case descendantCurrency = "-currency"
+        case descendantType = "-type"
+    }
+
+    override func toQuery() -> [String: String] {
+        var query = super.toQuery()
+        if let date = createdAfter {
+            query[QueryParam.createdAfter.rawValue] = HyperwalletTransferMethodQueryParam.iso8601.string(from: date)
+        }
+        if let date = createdBefore {
+            query[QueryParam.createdBefore.rawValue] = HyperwalletTransferMethodQueryParam.iso8601.string(from: date)
+        }
+        if let currency = currency {
+            query[QueryParam.currency.rawValue] = currency
+        }
+        if let sortBy = sortBy {
+            query[QueryParam.sortBy.rawValue] = sortBy.rawValue
+        }
+        return query
+    }
+
+    static let iso8601: DateFormatter = {
```

Comment:
Create an Extension of Date and move this logic to a func in Date extension. Use that func in here and HyperwalletTransferMethodPagination class

Reply:
fixed